processing/src/main/java/io/druid/segment/filter/EqualColumnsFilter.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.filter;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import io.druid.collections.bitmap.ImmutableBitmap;
+import io.druid.query.ColumnSelectorPlus;
+import io.druid.query.dimension.DefaultDimensionSpec;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.query.filter.BitmapIndexSelector;
+import io.druid.query.filter.Filter;
+import io.druid.query.filter.ValueGetter;
+import io.druid.query.filter.ValueMatcher;
+import io.druid.query.filter.ValueMatcherColumnSelectorStrategy;
+import io.druid.query.filter.ValueMatcherColumnSelectorStrategyFactory;
+import io.druid.segment.ColumnSelector;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.DimensionHandlerUtils;
+
+/**
+ */
+public class EqualColumnsFilter implements Filter
+{
+  private static final Joiner IS_JOINER = Joiner.on(" = ");
+
+  private final String[] dimensions;
+  private final ExtractionFn extractionFn;
+
+  public EqualColumnsFilter(
+      final String[] dimensions,
+      final ExtractionFn extractionFn
+  )
+  {
+    this.dimensions = Preconditions.checkNotNull(dimensions, "dimensions");
+    this.extractionFn = extractionFn;
```

Comment:
@jon-wei, I think you're right that this bears little resemblance to the rest of "selector" and makes more sense as a different DimFilter class. I agree with that.

On using DimensionSpecs instead of dimension names + an extractionFn, I could take or leave that one, so I would be ok with this patch either way. I'm ok with leaving that until someone actually has the need for it.